### ISSUE TITLE
Cannot resolve config property referenced in dev-console card template

### ISSUE THREAD

**Issue by knutwannheden** [quarkusio/quarkus#21691]
### Describe the bug

I have an extension contributing a card to the dev-console. The card's `embedded.html` contains a hyperlink to an endpoint contributed by the extension like so:
```html
<a href="{config:property('quarkus.http.root-path')}foo/bar" class="badge badge-light">...</a>
```

I would have to downgrade and test again, but I am fairly certain that this used to work properly prior to Quarkus 2.5.0.

Now with Quarkus 2.5.0 the config property `quarkus.http.root-path` can no longer be resolved by the template expression `config:property('quarkus.http.root-path')`.

I did a little bit of debugging and noticed that there are a lot of `SmallRyeConfig` objects being created. A separate one is also created once the dev-console is opened and this config object does not contain the default value for `quarkus.http.root-path` and it looks like it doesn't either contain the config source for the `application.properties` file.

### Expected behavior

The `embedded.html` file for a dev-console card should be able to reference run-time config properties (and build-time config properties visible at run-time) using template expressions like `config:property('quarkus.http.root-path')`. The expression should be resolved to the value provided by the application (`application.properties`, system properties, etc.) or the default value specified by the corresponding `@BuildItem` annotated field.

### Actual behavior

The `config:property()` template expression currently doesn't appear to be working properly. I get the impression that it may be using the wrong `SmallRyeConfig` instance. The whole expression is resolved to the empty string and I couldn't see any error being logged anywhere either.

### How to Reproduce?

Creating a reproducer would cost me some time, but I could look into it.

Basically all I did was to implement a Quarkus extension where I contribute a dev-console card with an `embedded.html` file as described above.

### Quarkus version or git rev

2.5.0.Final

**Comment by mkouba** [comment]
@knutwannheden Hm, I've just tried to modify the `embedded.html` in the qute extension and it renders correctly the value `/` which is the default value for `quarkus.http.root-path`.

**Comment by knutwannheden** [comment]
Hmm... I have to test this again. Possibly I ran my test in an multi-module Maven project where the Quarkus extension was part of the same Maven reactor build as the test project. Could that be a problem?

**Comment by mkouba** [comment]
> Hmm... I have to test this again. Possibly I ran my test in an multi-module Maven project where the Quarkus extension was part of the same Maven reactor build as the test project. Could that be a problem?

I have no idea :-). Anyway, let me know if it turns out to be a multi-module bug.

**Comment by knutwannheden** [comment]
Yes, you are right: The default value is properly found and returned. However, what doesn't work is that if I modify the default (e.g. using `application.properties`) then this value is not found and returned. As a result my extension's card will end up using the HTTP root `/` rather than e.g. `/foo`.

**Comment by mkouba** [comment]
> Yes, you are right: The default value is properly found and returned. However, what doesn't work is that if I modify the default (e.g. using `application.properties`) then this value is not found and returned. As a result my extension's card will end up using the HTTP root `/` rather than e.g. `/foo`.

You're right. I can confirm that the default value is used.

So the config namespace resolver is using `ConfigProvider.getConfig().getOptionalValue()` which is probably wrong? I can see that the config editor page is built on top of `List<ConfigDescriptionBuildItem>`. Maybe we should switch to these build items as well? @radcortez @phillip-kruger WDYT?

**Comment by radcortez** [comment]
Hum, if you see multiple `SmallRyeConfig` instances being created it means that the lookup was done on the wrong classloader so a new one is created to the current classloader. That means that the configuration may differ completely.

The`ConfigDescriptionBuildItem` does not carry the config value resolution. Just metadata like the config description that the regular `Config` instance does not have.

Let me have a look into it.

**Comment by knutwannheden** [comment]
@radcortez Yes, that was also my thinking. And like I said, the instance being used here is created at the point when the dev console is first accessed.

**Comment by mkouba** [comment]
> The`ConfigDescriptionBuildItem` does not carry the config value resolution.

Hmm, you're right. The value is populated lazily, when the table of config editor is loaded and it also uses `ConfigProvider.getConfig()`. The difference is that we register the value resolver in a `io.quarkus.vertx.http.deployment.devmode.console.DevConsoleProcessor` method, i.e. using a different TCCL. I'll try to fix it.
